snow/validators/retriever_test.go

```diff
@@ -0,0 +1,92 @@
+package validators
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/utils/constants"
+)
+
+func TestCachingRetriever_GetValidators(t *testing.T) {
+	testBlockID := ids.ID{
+		0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,
+		0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+		0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F,
+	}
+
+	t.Run("cache call count", func(t *testing.T) {
+		t.Parallel()
+
+		callsCount := 0
+		cacheCallCount := 0
+		validators := make(map[ids.ID]Set)
+		retrieverMock := &TestRetriever{
+			GetValidatorsByBlockIDFunc: func(blockID ids.ID) (Set, error) {
+				callsCount++
+
+				if set, ok := validators[blockID]; ok {
+					cacheCallCount++
+					return set, nil
+				}
+				validators[blockID] = loadCostonValidators()
+				return validators[blockID], nil
+			},
+		}
+		cr := NewCachingRetriever(retrieverMock)
+		set, err := cr.GetValidators(testBlockID)
+		assert.NoError(t, err)
+		assert.Equal(t, loadCostonValidators(), set)
+		assert.Equal(t, 1, callsCount)
+		assert.Equal(t, 0, cacheCallCount)
+
+		set, err = cr.GetValidators(testBlockID)
+		assert.Equal(t, 1, callsCount)
+		assert.Equal(t, 0, cacheCallCount)
+		assert.Equal(t, uint64(1000000), set.Weight())
```

Comment:
Shouldn't we also check for an error here?

```suggestion
		set, err = cr.GetValidators(testBlockID)
		require.NoError(t, err)
		assert.Equal(t, 1, callsCount)
		assert.Equal(t, 0, cacheCallCount)
		assert.Equal(t, uint64(1000000), set.Weight())
```